packages/devtools_app/lib/src/screens/debugger/codeview.dart

```diff
@@ -544,6 +581,174 @@ class _CodeViewState extends State<CodeView>
+                          'Self %',
+                          textAlign: TextAlign.center,
+                        ),
+                      ),
+                    ),
+                  ],
+                ),
+              ),
+              const Divider(
+                height: 0,
+              ),
+              Expanded(
+                child: ListView.builder(
+                  controller: scrollController,
+                  itemExtent: CodeView.rowHeight,
+                  itemCount: lineCount,
+                  itemBuilder: (context, index) {
+                    final lineNum = lineOffset + index + 1;
+                    final data = sourceReport.profilerEntries[lineNum];
+                    if (data == null) {
+                      return const SizedBox();
+                    }
+                    return ProfileInformationGutterItem(
+                      lineNumber: lineNum,
+                      profilerData: data,
+                    );
+                  },
+                ),
+              ),
+            ],
+          ),
+          const Center(
+            child: VerticalDivider(),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class ProfileInformationGutterItem extends StatelessWidget {
+  const ProfileInformationGutterItem({
+    Key? key,
+    required this.lineNumber,
+    required this.profilerData,
+  }) : super(key: key);
+
+  final int lineNumber;
+
+  final ProfileReportEntry profilerData;
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      height: CodeView.rowHeight,
+      child: Row(
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        children: [
+          Expanded(
+            child: ProfilePercentageItem(
+              percentage: profilerData.inclusivePercentage,
+              hoverText: ProfileInformationGutter.totalTimeTooltip,
+            ),
+          ),
+          Expanded(
+            child: ProfilePercentageItem(
+              percentage: profilerData.exclusivePercentage,
+              hoverText: ProfileInformationGutter.selfTimeTooltip,
+            ),
+          ),
+        ],
+      ),
+    );
+  }
+}
+
+class ProfilePercentageItem extends StatelessWidget {
+  const ProfilePercentageItem({
+    required this.percentage,
+    required this.hoverText,
+  });
+
+  final double percentage;
+  final String hoverText;
+
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    Color? color;
+    if (percentage > 5) {
+      color = colorScheme.performanceHighImpactColor;
+    } else if (percentage > 1) {
+      color = colorScheme.performanceMediumImpactColor;
+    } else {
+      color = colorScheme.performanceLowImpactColor;
+    }
+    return DevToolsTooltip(
+      message: hoverText,
+      child: Container(
+        color: color,
+        child: Container(
+          margin: const EdgeInsets.symmetric(
+            horizontal: densePadding,
+          ),
```

Comment:
merge these two containers. If you change margin to padding, it will have the same effect